src/OpenLoco/src/Paint/PaintWall.cpp

```diff
@@ -0,0 +1,226 @@
+            { 5, 3, 1 },
+            { 4, 2, 0 },
+        };
+
+        // TODO: Add the appropriate getters to WallElement
+        uint8_t type = elWall.rawData()[0];
+
+        // TODO: Make this branch-less.
+        uint32_t imageOffset = kImageOffsets[rotation][0];
+        if ((type & 0x80u) == 0)
+        {
+            imageOffset = kImageOffsets[rotation][1];
+            if ((type & 0x40) == 0)
+                imageOffset = kImageOffsets[rotation][2];
+        }
+
+        return imageOffset;
+    }
+
+    static ImageId getWallImageId(ImageIndex imageIndex, bool isGhost, const World::WallElement& elWall, const WallObject* wallObject)
+    {
+        if (isGhost)
+        {
+            return Gfx::applyGhostToImage(imageIndex);
+        }
+
+        ImageId imageId{ imageIndex };
+
+        if ((wallObject->flags & WallObjectFlags::hasPrimaryColour) != WallObjectFlags::none)
+        {
+            imageId = imageId.withPrimary(elWall.getPrimaryColour());
+        }
+
+        if ((wallObject->flags & WallObjectFlags::hasSecondaryColour) != WallObjectFlags::none)
+        {
+            imageId = imageId.withSecondary(elWall.getSecondaryColour());
+        }
+
+        return imageId;
+    }
+
+    // 0x004C3D7C
+    void paintWall(PaintSession& session, const World::WallElement& elWall)
+    {
+        const auto* wallObject = ObjectManager::get<WallObject>(elWall.wallObjectId());
+        assert(wallObject != nullptr);
+
+        const auto isGhost = elWall.isGhost();
+        session.setItemType(isGhost ? InteractionItem::noInteraction : InteractionItem::wall);
+
+        uint32_t frameNo = 0;
+        if (wallObject->var_09 & 0x10)
+        {
+            frameNo = 2 * (ScenarioManager::getScenarioTicks() & 7U);
+        }
+
+        const coord_t height = 4 * wallObject->height - 2;
+        const coord_t baseHeight = elWall.baseHeight();
+        const coord_t baseHeightEnd = baseHeight + 1;
+        const int32_t rotation = (session.getRotation() + elWall.rotation()) & 0x3;
+
+        uint32_t imageOffset = 0;
+        uint32_t imageIndex = 0;
+
+        switch (rotation)
+        {
+            case 0:
+                imageOffset = getWallImageIndexOffset(elWall, rotation);
+                imageIndex = frameNo + wallObject->sprite + imageOffset;
+                if ((wallObject->flags & WallObjectFlags::unk1) != WallObjectFlags::none)
+                {
+                    const auto imageId = getWallImageId(imageIndex, isGhost, elWall, wallObject);
+                    session.addToPlotListAsParent(
+                        imageId,
+                        World::Pos3{ 0, 0, baseHeight },
+                        World::Pos3{ 1, 1, baseHeightEnd },
+                        World::Pos3{ 1, 28, height });
+
+                    if (!isGhost)
+                    {
+                        session.addToPlotListAsChild(
+                            imageId.withRemap(ExtColour::unk74).withIndexOffset(6),
```

Comment:
this isn't correct it would be:
`imageId.withTranslucency(ExtColour::unk74 + imageId.getPrimaryColour()).withIndexOffset(6);`
this is because its + 0x23A00006 so remap + remap == translucency. Which also likely means i've made a mistake in paint vehicle... as i see almost the same code there.

Reply:
Changed it, you are correct.